tachyon/math/polynomials/univariate/univariate_evaluations.h

```diff
@@ -36,49 +36,49 @@ class UnivariateEvaluationsOp;
 // evaluation domain, the univariate polynomial can vary.
 // For more information, refer to
 // https://en.wikipedia.org/wiki/Lagrange_polynomial
-template <typename F, size_t MaxDegree>
+template <typename F, size_t N>
 class UnivariateEvaluations final
-    : public Polynomial<UnivariateEvaluations<F, MaxDegree>> {
+    : public Polynomial<UnivariateEvaluations<F, N>> {
  public:
-  constexpr static size_t kMaxDegree = MaxDegree;
+  constexpr static size_t kMaxSize = N;
 
   using Field = F;
 
   constexpr UnivariateEvaluations() : UnivariateEvaluations({F::Zero()}) {}
   constexpr explicit UnivariateEvaluations(const std::vector<F>& evaluations)
       : evaluations_(evaluations) {
-    CHECK_LE(Degree(), MaxDegree);
+    CHECK_LE(evaluations_.size(), kMaxSize);
```

Comment:
`UnivariateEvaluations` does not add or remove leading zeros, so it must receive `evaluations` of the exact length. And if `kMaxSize` represents 2^s, then the formula is correct, but if it means 2^n, we should change it to `CHECK_EQ`. 

In this PR, we interpret `kMaxSize` as 2^s, and I will correct it in the PR that assigns Size to PCS.

Reply:
You can resolve this by yourself!